src/mlpack/methods/ann/layer/lookup_impl.hpp

```diff
@@ -21,30 +21,39 @@ namespace ann /** Artificial Neural Network. */ {
 
 template <typename InputDataType, typename OutputDataType>
 Lookup<InputDataType, OutputDataType>::Lookup(
-    const size_t inSize,
-    const size_t outSize) :
-    inSize(inSize),
-    outSize(outSize)
+    const size_t vocabSize,
+    const size_t embeddingSize) :
+    vocabSize(vocabSize),
+    embeddingSize(embeddingSize)
 {
-  weights.set_size(outSize, inSize);
+  weights.set_size(vocabSize, embeddingSize);
 }
 
 template<typename InputDataType, typename OutputDataType>
 template<typename eT>
 void Lookup<InputDataType, OutputDataType>::Forward(
     const arma::Mat<eT>& input, arma::Mat<eT>& output)
 {
-  output = weights.cols(arma::conv_to<arma::uvec>::from(input) - 1);
+  const size_t seqLength = input.n_rows;
+  const size_t batchSize = input.n_cols;
+
+  output.set_size(seqLength * embeddingSize, batchSize);
+
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    output.col(i) = arma::vectorise(weights.rows(
```

Comment:
Could you add a comment somewhere to describe why this ordering has been chosen? Some info from https://github.com/mlpack/mlpack/pull/2398#issuecomment-667671075 will do.

Reply:
@lozhnikov: Is this comprehensible? I don't know if a third person reading this would understand or not 😬. This reminded me of a [tweet](https://pasteboard.co/JkCofMi.png) I read sometimes back 😂

Reply:
I think you could just point out that the multi-head attention layer requires this particular ordering.